solver/src/main/java/org/chocosolver/solver/search/strategy/selectors/variables/DomOverWDeg.java

```diff
@@ -99,15 +215,92 @@ final int remapInc() {
 
     @Override
     public void afterRestart() {
-        /*if (vars[0].getModel().getSolver().getSolutionCount() > solution) {
-            solution = vars[0].getModel().getSolver().getSolutionCount();
+        if (flushWeights()) {
+            weights.clear();
+        }
+    }
+
+
+    /// ///////////////////////////////////////////////////////////////////
+    /// /////////////// THIS IS RELATED TO INCREMENTAL FUTVARS ////////////
+    /// ///////////////////////////////////////////////////////////////////
+
+    private final BiFunction<Variable, Integer, Integer> inc = (v, i) -> i + 1;
+
+    final void plug(Variable var) {
+        if (!observed.containsKey(var)) {
+            observed.put(var, 1);
+            var.addMonitor(this);
+        } else {
+            observed.computeIfPresent(var, inc);
```

Comment:
Why not call directly `observed.compute(var, inc)` ? `var` is a key here, and the associated value is not null "by construction" of the `observed` map